staging/src/k8s.io/apiserver/pkg/endpoints/handlers/apply/gvkparser.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apply
```

Comment:
Can we put these in a subpackage?

Reply:
It actually makes sense in this package since this package is about converting ManagedFields from/to API to/from smd.  But I agree that the package name is pretty bad.  We'll rename the package in a follow-up PR if you don't mind.